Include Ad Hoc Step for OnPrem 2013 (#3298)
MicrosoftDocs/OfficeDocs-SharePoint: SharePoint/SharePointServer/hybrid/configure-server-to-server-authentication.md

5. After script execution, users will not see any changes when this change is implemented.

#### Step 8 (Only required for Sharepoint On-Premises 2013): Give New App Principal QueryAsUserIgnoreAppPrincipal permission
<a name="step10"> </a>

> The on-premises version of Sharepoint 2013 needs a hidden constraint in every federated query. The reverse proxy returns the documents indexed in the reverse proxy site itself, not the internal on-premise search site as expected. To avoid this, you need to execute the following steps in your Sharepoint on-premises 2013 admin site:

Reviewer: ```suggestion
The on-premises version of Sharepoint 2013 needs a hidden constraint in every federated query. The reverse proxy returns the documents indexed in the reverse proxy site itself, not the internal on-premise search site as expected. To avoid this, you need to execute the following steps in your Sharepoint 2013 on-premises admin site:
```

> The on-premises version of Sharepoint 2013 needs a hidden constraint in every federated query. The reverse proxy returns the documents indexed in the reverse proxy site itself, not the internal on-premise search site as expected. To avoid this, you need to execute the following steps in your Sharepoint 2013 on-premises admin site:

Reviewer: ```suggestion
The on-premises version of SharePoint 2013 needs a hidden constraint in every federated query. The reverse proxy returns the documents indexed in the reverse proxy site itself, not the internal on-premise search site as expected. To avoid this, you need to execute the following steps in your SharePoint 2013 on-premises admin site:
```

1. Go to <CentralAdminURL>/_layouts/appinv.aspx and Search for **c3959f3a-5ad4-4d2b-b1f0-bc70f9a5d0a1**, where you should find **Greenland Federated Search Bot Skill**.

2. If there are items in the App Domain field, leave them be, and if it is empty, use localhost.
  
3. In the Redirect URL, use https://localhost.
  

> 4. In the Permission Request XML field paste the following XML excerpt:

Reviewer: ```suggestion
4. In the Permission Request XML field, paste the following XML excerpt:
```

> 2. If there are items in the App Domain field, leave them be, and it is empty, use localhost.

Reviewer 2: looks like **if** is missing, should it be 'and **if** it is empty'

Author: Fixed.

  
  ```xml
<AppPermissionRequests>
<AppPermissionRequest Scope="http://sharepoint/search" Right="QueryAsUserIgnoreAppPrincipal" />
</AppPermissionRequests>
  ```

> 5. The configuration page should look as in the screenshot below, finally, click on Create.

Reviewer: ```suggestion
5. The configuration page should appear similar to the following screenshot. Finally, select **Create**.
```
